osu.Game.Tests/Visual/Settings/TestSceneSettingsPanel.cs

```diff
@@ -34,6 +35,29 @@ public void SetUpSteps()
             });
         }
 
+        [Test]
+        public void TestQuickFiltering()
+        {
+            AddStep("set filter", () =>
+            {
+                settings.SectionsContainer.ChildrenOfType<SearchTextBox>().First().Current.Value = "scaling";
+            });
+
+            AddUntilStep("wait for items to load", () => settings.SectionsContainer.ChildrenOfType<IFilterable>().Any());
+
+            AddAssert("ensure all items match filter", () => settings.SectionsContainer
+                                                                     .ChildrenOfType<SettingsSection>().Where(f => f.IsPresent)
+                                                                     .All(section =>
+                                                                         section.Children.Where(f => f.IsPresent)
+                                                                                .OfType<ISettingsItem>()
+                                                                                .OfType<IFilterable>()
+                                                                                .Where(f => !(f is IHasFilterableChildren))
+                                                                                .All(f => f.FilterTerms.Any(t => t.Contains("scaling")))
+                                                                     ));
+
+            AddAssert("ensure section is current", () => settings.CurrentSection.Value != null);
```

Comment:
For solidifying the assertion, as I had the current one false-passing while I was looking at the scheduling of `InvalidateScrollPosition` from above.
```suggestion
            AddAssert("ensure section is current", () => settings.CurrentSection.Value == settings.SectionsContainer.ChildrenOfType<SettingsSection>().First(s => s.IsPresent));
```

Reply:
Which section was selected in the scenario you ran in to?

Reply:
Before the recent filter event changes, I was playing with the way `InvalidateScrollPosition` is scheduled and ended up with `settings.CurrentSection.Value` returning the first section in the settings panel ("General", which is not present), making the assertion incorrectly pass.